tests/eips/eip4844/test_excess_data_gas.py

```diff
@@ -233,28 +236,31 @@ def blocks(  # noqa: D103
     tx: Transaction,
     header_excess_data_gas: Optional[int],
     header_data_gas_used: Optional[int],
+    block_intermediate: Block,
```

Comment:
Directly related to the comment on `common.py`, we can check if the `block_intermediate` is `None`, and only add it to the start of the block list if it contains a value.